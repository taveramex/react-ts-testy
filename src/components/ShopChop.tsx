import { ItemCounter, ItemCounterProps } from '../shopping-cart/ItemCounter'

const products:ItemCounterProps[] = [
  {id:1, productName:'Perrito Kawaii', quantity:7, price:10000},
  {id:2, productName:'Loro', quantity:1, price:20},
  {id:3, productName:'Gato Gordo generico', quantity:1, price:630},
  {id:4, productName:'Kuckee lorito', quantity:34, price:2400},
  {id:5, productName:'rock', quantity:9999, price:12},
  {id:6, productName:'Pez', quantity:231, price:15},
]

const ShopChop = () => {
  return (
    <>
      <h1>welcome to some shady shop to chop</h1>
      <p>heres our products</p>

      { products.map(({id:key, productName:name, quantity}) => (<ItemCounter key={key} productName={name} quantity={quantity} /> )) }
    
    </>
  )
}

export default ShopChop;
