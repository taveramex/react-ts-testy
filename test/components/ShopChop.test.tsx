import { describe, test, expect, vi, afterEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import ShopChop from '../../src/components/ShopChop';

const mockItemCounter = vi.fn((props: unknown) => (<div data-testid="ItemCounter" />));

vi.mock('../../src/shopping-cart/ItemCounter', (props: unknown) => ({
    ItemCounter: (props: unknown) => mockItemCounter(props),
}));

describe("ShopChop", () => {
    afterEach(()=>{
        vi.clearAllMocks();
    });
    test("should Match screenshot", () => {
        const { container } = render(<ShopChop />);
        expect(container).toMatchSnapshot();
    });
    test("should render the correct number of ItemCounter components", () => {
        render(<ShopChop />);
        const items = screen.getAllByTestId("ItemCounter");
        expect(items.length).toBe(6);
    });
    test("shoud render itemCounter with correct props", ()=>{
        render(<ShopChop />);

        expect(mockItemCounter).toHaveBeenCalledTimes(6);
        expect(mockItemCounter).toHaveBeenCalledWith(
            {productName:'Gato Gordo generico', quantity:1,}
        );

    });
});