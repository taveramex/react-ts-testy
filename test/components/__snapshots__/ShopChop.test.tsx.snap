// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShopChop > should Match screenshot 1`] = `
<div>
  <h1>
    welcome to some shady shop to chop
  </h1>
  <p>
    heres our products
  </p>
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
</div>
`;
